config/config_test.go

```diff
@@ -155,3 +168,131 @@ func newConfig(conf resource.Config) (*Config, error) {
+		// Test for missing camera name attribute
+		logger.Debug("Testing SLAM config without camera[name]")
+		cfgService := makeCfgService(true)
+		delete(cfgService.Attributes["camera"].(map[string]string), "name")
+		_, err := newConfig(cfgService)
+		test.That(t, err, test.ShouldBeError, newError(utils.NewConfigValidationFieldRequiredError(testCfgPath, "camera[name]").Error()))
+	})
+
+	t.Run("Config with invalid parameter type with feature flag enabled", func(t *testing.T) {
+		cfgService := makeCfgService(true)
+		cfgService.Attributes["data_dir"] = true
+		_, err := newConfig(cfgService)
+		expE := newError("1 error(s) decoding:\n\n* 'data_dir' expected type 'string', got unconvertible type 'bool', value: 'true'")
+		test.That(t, err, test.ShouldBeError, expE)
+		cfgService.Attributes["data_dir"] = "true"
+		_, err = newConfig(cfgService)
+		test.That(t, err, test.ShouldBeNil)
+	})
+
+	t.Run("Config with invalid camera data_frequency_hz with feature flag enabled", func(t *testing.T) {
+		cfgService := makeCfgService(true)
+		cfgService.Attributes["camera"] = map[string]string{
+			"name":              "a",
+			"data_frequency_hz": "twenty",
+		}
+		_, err := newConfig(cfgService)
+		test.That(t, err, test.ShouldBeError, newError("camera[data_frequency_hz] must only contain digits"))
+	})
+
+	t.Run("Config with out of range values with feature flag enabled", func(t *testing.T) {
+		cfgService := makeCfgService(true)
+		cfgService.Attributes["camera"] = map[string]string{
+			"name":              "a",
+			"data_frequency_hz": "-1",
+		}
+		_, err := newConfig(cfgService)
+		test.That(t, err, test.ShouldBeError, newError("cannot specify camera[data_frequency_hz] less than zero"))
+		cfgService.Attributes["camera"] = map[string]string{
+			"name":              "a",
+			"data_frequency_hz": "1",
+		}
+		cfgService.Attributes["map_rate_sec"] = -1
+		_, err = newConfig(cfgService)
+		test.That(t, err, test.ShouldBeError, newError("cannot specify map_rate_sec less than zero"))
+	})
+
+	t.Run("All parameters e2e with feature flag enabled", func(t *testing.T) {
+		cfgService := makeCfgService(true)
+		cfgService.Attributes["camera"] = map[string]string{
+			"name":              "a",
+			"data_frequency_hz": "20",
+		}
+		cfgService.Attributes["map_rate_sec"] = 1002
+
+		cfgService.Attributes["config_params"] = map[string]string{
+			"mode":    "test mode",
+			"value":   "0",
+			"value_2": "test",
+		}
+		cfg, err := newConfig(cfgService)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, cfg.DataDirectory, test.ShouldEqual, cfgService.Attributes["data_dir"])
+		test.That(t, cfg.Camera, test.ShouldResemble, cfgService.Attributes["camera"])
+		test.That(t, *cfg.MapRateSec, test.ShouldEqual, cfgService.Attributes["map_rate_sec"])
+		test.That(t, cfg.ConfigParams, test.ShouldResemble, cfgService.Attributes["config_params"])
+	})
+}
+
+func TestGetOptionalParametersFeatureFlag(t *testing.T) {
```

Comment:
Can we please have a test case of which returns an error, now that this function can possibly return an error?

Reply:
Turns out we already check for it being convertible in validate, so I just took out the error from the function